### ISSUE TITLE
optirun doesn't work, but with empty error message

### ISSUE THREAD

**Issue by leledumbo** [Bumblebee-Project/Bumblebee#573]
Distro: Manjaro Linux 0.8.9

Kernel: Linux manjaro 3.14.0-1-MANJARO #1 SMP PREEMPT Sat Apr 5 00:44:46 UTC 2014 x86_64 GNU/Linux

Manufacturer and model:
Hewlett-Packard
baseboard-product-name: 2165
baseboard-version     : 29.23
system-manufacturer   : Hewlett-Packard
system-product-name   : HP Pavilion 14 Notebook PC
system-version        : 0887100000005E00000600000
bios-vendor           : Insyde
bios-version          : F.21
bios-release-date     : 08/08/2013

Bumblebee version: 3.2.1

```
$ optirun -vv glxinfo    
[15906.019158] [DEBUG]Reading file: /etc/bumblebee/bumblebee.conf
[15906.019691] [INFO]Configured driver: nvidia
[15906.019925] [DEBUG]optirun version 3.2.1 starting...
[15906.019957] [DEBUG]Active configuration:
[15906.019972] [DEBUG] bumblebeed config file: /etc/bumblebee/bumblebee.conf
[15906.019988] [DEBUG] X display: :8
[15906.020002] [DEBUG] LD_LIBRARY_PATH: /usr/lib/nvidia:/usr/lib32/nvidia
[15906.020017] [DEBUG] Socket path: /var/run/bumblebee.socket
[15906.020032] [DEBUG] Accel/display bridge: auto
[15906.020046] [DEBUG] VGL Compression: proxy
[15906.020060] [DEBUG] VGLrun extra options: 
[15906.020074] [DEBUG] Primus LD Path: /usr/lib/primus:/usr/lib32/primus
[15906.020163] [DEBUG]Using auto-detected bridge primus
[15906.025603] [INFO]Response: No - error: [ XORG ] (EE)

[15906.025634] [ERROR]Cannot access secondary GPU - error: [ XORG ] (EE)

[15906.025642] [DEBUG]Socket closed.
[15906.025670] [ERROR]Aborting because fallback start is disabled.
[15906.025680] [DEBUG]Killing all remaining processes.
```

As can be seen, I have no error message to indicate what went wrong, thus I have no idea what to fix. My nVidia card is GT 740M.

Following is the content of my xorg.conf.nvidia:

```
##
## Generated by mhwd - Manjaro Hardware Detection
##


Section "ServerLayout"
    Identifier "Layout0"
    Option "AutoAddDevices" "false"
EndSection

Section "Device"
    Identifier  "Device1"
    Driver      "nvidia"
    VendorName "NVIDIA Corporation"
    Option "NoLogo" "true"
    Option "UseEDID" "false"
    Option "ConnectedMonitor" "DFP"
    BusID "PCI:A:0:0"
EndSection
```

xorg.conf.nouveau:

```
Section "ServerLayout"
    Identifier  "Layout0"
    Option      "AutoAddDevices" "false"
    Option      "AutoAddGPU" "false"
EndSection

Section "Device"
    Identifier  "DiscreteNvidia"
    Driver      "nouveau"

#   If the X server does not automatically detect your VGA device,
#   you can manually set it here.
#   To get the BusID prop, run `lspci | egrep 'VGA|3D'` and input the data
#   as you see in the commented example.
#   This Setting is needed on Ubuntu 13.04.
#   BusID "PCI:01:00:0"

    BusID "PCI:0:2:0"
EndSection
```

and bumblebee.conf:

```
# Configuration file for Bumblebee. Values should **not** be put between quotes

## Server options. Any change made in this section will need a server restart
# to take effect.
[bumblebeed]
# The secondary Xorg server DISPLAY number
VirtualDisplay=:8
# Should the unused Xorg server be kept running? Set this to true if waiting
# for X to be ready is too long and don't need power management at all.
KeepUnusedXServer=false
# The name of the Bumbleblee server group name (GID name)
ServerGroup=users
# Card power state at exit. Set to false if the card shoud be ON when Bumblebee
# server exits.
TurnCardOffAtExit=false
# The default behavior of '-f' option on optirun. If set to "true", '-f' will
# be ignored.
NoEcoModeOverride=false
# The Driver used by Bumblebee server. If this value is not set (or empty),
# auto-detection is performed. The available drivers are nvidia and nouveau
# (See also the driver-specific sections below)
Driver=nvidia
# Directory with a dummy config file to pass as a -configdir to secondary X
XorgConfDir=/etc/bumblebee/xorg.conf.d

## Client options. Will take effect on the next optirun executed.
[optirun]
# Acceleration/ rendering bridge, possible values are auto, virtualgl and
# primus.
Bridge=auto
# The method used for VirtualGL to transport frames between X servers.
# Possible values are proxy, jpeg, rgb, xv and yuv.
VGLTransport=proxy
# List of paths which are searched for the primus libGL.so.1 when using
# the primus bridge
PrimusLibraryPath=/usr/lib/primus:/usr/lib32/primus
# Should the program run under optirun even if Bumblebee server or nvidia card
# is not available?
AllowFallbackToIGC=false


# Driver-specific settings are grouped under [driver-NAME]. The sections are
# parsed if the Driver setting in [bumblebeed] is set to NAME (or if auto-
# detection resolves to NAME).
# PMMethod: method to use for saving power by disabling the nvidia card, valid
# values are: auto - automatically detect which PM method to use
#         bbswitch - new in BB 3, recommended if available
#       switcheroo - vga_switcheroo method, use at your own risk
#             none - disable PM completely
# https://github.com/Bumblebee-Project/Bumblebee/wiki/Comparison-of-PM-methods

## Section with nvidia driver specific options, only parsed if Driver=nvidia
[driver-nvidia]
# Module name to load, defaults to Driver if empty or unset
KernelDriver=nvidia
PMMethod=auto
# colon-separated path to the nvidia libraries
LibraryPath=/usr/lib/nvidia:/usr/lib32/nvidia
# comma-separated path of the directory containing nvidia_drv.so and the
# default Xorg modules path
XorgModulePath=/usr/lib/nvidia/xorg/,/usr/lib/xorg/modules
XorgConfFile=/etc/bumblebee/xorg.conf.nvidia

## Section with nouveau driver specific options, only parsed if Driver=nouveau
[driver-nouveau]
KernelDriver=nouveau
PMMethod=auto
XorgConfFile=/etc/bumblebee/xorg.conf.nouveau
```

Last but not least, my related lspci output:

```
00:02.0 VGA compatible controller: Intel Corporation Haswell-ULT Integrated Graphics Controller (rev 09)
0a:00.0 3D controller: NVIDIA Corporation GK208M [GeForce GT 740M] (rev a1)
```

Tarball: [bumblebee-bugreport-20140415_000544.tar.gz](https://gist.githubusercontent.com/Lekensteyn/6211401/raw/2a8474bba9551d791040a2ffcd5aa13c3a65f16f/bumblebee-bugreport-20140415_000544.tar.gz)

**Comment by Lekensteyn** [comment]
@ArchangeGabriel this bug was already closed, what else needs to be done here?

**Comment by ArchangeGabriel** [comment]
Nothing, this was just to say that this was going to be fixed in 4.0, and that you took care of it. ;)

**Comment by Lekensteyn** [comment]
Alrighty, another thing off the todo list :)
